app/controllers/api/maintenance_stats_controller.rb

```diff
@@ -19,31 +19,49 @@ def begin_watching_bulk
     head :accepted
   end
 
+  # A possible roadmap for the future is to replace this endpoint with a "track_repository" that:
+  #  - creates the repository if we haven't (so we don't have to track everything just in case)
+  #  - marks it interesting, even if it isn't github
+  #  - starts watching maint stats if supported (if it is github, for now)
   def begin_watching_repositories
     lookup_names = params.require(:repositories).map { |p| p.permit(%i[host_type full_name]).to_h.symbolize_keys }
 
-    supported_host_types = ["github"]
+    github_lookup_names, other_lookup_names = lookup_names.partition { |repo_host_and_name| repo_host_and_name[:host_type].downcase == "github" }
 
-    # we only support GitHub so grab a working token now so we
-    # don't run into issues trying to find one in the loop for
-    # each repository name
-    auth_token = AuthToken.find_token(:v3)
+    # grab a working token now so we don't run into issues trying to find one in the loop for each
+    # repository name
+    auth_token = AuthToken.find_token(:v3) if github_lookup_names.present?
 
-    repos = lookup_names.map do |repo_host_and_name|
-      return render json: { error: "#{repo_host_and_name[:host_type]} is not a supported host" }, status: :bad_request unless supported_host_types.include?(repo_host_and_name[:host_type].downcase)
+    # do we really want to synchronously create and refresh these repos here?
+    # Not sure this is the right choice.
 
+    github_repos = github_lookup_names.map do |repo_host_and_name|
       Repository.create_from_host(repo_host_and_name[:host_type], repo_host_and_name[:full_name], auth_token.token)
     end
 
+    other_repos = other_lookup_names.map do |repo_host_and_name|
+      Repository.create_from_host(repo_host_and_name[:host_type], repo_host_and_name[:full_name], nil)
+    end
+
```

Comment:
so currently here is how we find an existing repo, after we do the API call to look it up:
https://github.com/librariesio/libraries.io/blob/2c24c8fb1a4e852acde67923e26d25b55e87ea97/app/domain/repository/persist_repository_from_upstream.rb#L45-L50

I am wondering whether we should do this FIRST here, before we do the repo lookup, but I guess we would miss some name canonicalization and not have the uuid.

Reply:
yeah, that makes a lot of sense. The only reason I can think of to do the API lookup first is the "clash" removal part, where we'd destroy a found repo because the name didn't match, but I don't think we even reach that part because `find_repository_from_host_data` would find that repo in the first place.

Reply:
I changed it to do the lookup of existing repos first.